### PR TITLE
player as Entity

### PR DESCRIPTION
**Pull request by norgeous**

### CODE REVIEW

File: src/objects/entities/Entity.ts [hunk 1]
```diff
@@ -11,35 +10,30 @@ type AnimationsConfigType = {
...
 };
 
-type EntityConfigType = {
+export type EntityConfigType = {
   name: string;
   spriteSheetKey: string;
   animations: AnimationsConfigType[];
-  physicsConfig?: Phaser.Types.Physics.Matter.MatterSetBodyConfig;
-  keepUprightStratergy: KeepUprightStratergies;
+  physicsConfig?: MatterJS.IChamferableBodyDefinition & {
```
Comment (ScottyXIII) [inline src/objects/entities/Entity.ts:18]: I tried using `IChamferableBodyDefinition ` before but it doesn't have a type param. We were passing this to create the hitbox and type was used to tell it which shape for the collision, circle|rectangle and some others. 

Actually I'm not sure how that was working as `physicsConfig `is past to `Bodies.rectangle()` 🤔. It's fine for now but If we want diff collision shapes we'll have to do a check on `physicsConfig.type` then pass it to either `Bodies.circle()` or `Bodies.rectangle()`. There's a few other shapes as well.

File: src/objects/entities/Ben3.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  name: KEY,
+  spriteSheetKey: KEY,
+  facing: 1,
+  scale: 1,
+  craftpixOffset: {
+    x: 0,
+    y: 0,
+  },
+  physicsConfig: {
+    type: 'rectangle',
```
Comment (ScottyXIII) [inline src/objects/entities/Ben3.ts:24]: We can prob remove this for now.

File: src/objects/entities/Bat.ts [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
   scale: 1.5,
-  maxSpeedX: 1,
-  maxSpeedY: 2,
-  constantMotion: true,
+  craftpixOffset: {
+    x: 0,
+    y: 0,
+  },
   physicsConfig: {
     type: 'rectangle',
```
Comment (ScottyXIII) [inline src/objects/entities/Bat.ts:19]: can remove this for now.

File: src/objects/entities/Hedgehog.ts [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
   facing: -1,
   scale: 2,
-  maxSpeedX: 1,
-  maxSpeedY: 2,
+  craftpixOffset: {
+    x: 0,
+    y: 0,
+  },
   physicsConfig: {
     type: 'rectangle',
```
Comment (ScottyXIII) [inline src/objects/entities/Hedgehog.ts:19]: can remove this for now.

File: src/scenes/game-scene.ts [hunk 5]
```diff
@@ -134,15 +138,27 @@ class GameScene extends Phaser.Scene {
...
+    if (!this.parallax || !this.level || !this.player) return;
 
     this.parallax.update();
 
-    this.player = this.level.spawners.player.getChildren()[0] as Ben1;
-    smoothMoveCameraTowards(this, this.player.torso, 0.9);
+    smoothMoveCameraTowards(this, this.player.gameObject, 0.8);
+
+    const [myNum, setMyNum] = useLocalStorage('testNum', 0);
+    setMyNum(myNum + 1);
```
Comment (ScottyXIII) [inline src/scenes/game-scene.ts:161]: bit of Debugging aye?

File: src/scenes/game-scene.ts [hunk 6]
```diff
@@ -134,15 +138,27 @@ class GameScene extends Phaser.Scene {
     toggleMusic(this); // attaches listener to mute button
     const [isMute] = useLocalStorage('isMute', false);
     this.game.sound.mute = isMute; // set game mute to saved ls value
+
+    this.player = this.level.spawners.player.getChildren()[0] as Ben3;
+
+    const canvas = document.querySelector('#game');
+    canvas?.addEventListener('click', this.jump.bind(this));
```
Comment (ScottyXIII) [inline src/scenes/game-scene.ts:145]: Does click fire with touch on mobile as well? I'm not able to test over the network still :(

File: src/scenes/game-scene.ts (hunk 5 above)
Comment (norgeous) [reply src/scenes/game-scene.ts:161]: seems to be running well on every frame!
i just checked and my value is `6568819` - which is the number of frames i've rendered over lifetime of having this code here

File: src/scenes/game-scene.ts (hunk 6 above)
Comment (norgeous) [reply src/scenes/game-scene.ts:145]: yeah you're right... it **doesn't work**

File: src/scenes/game-scene.ts (hunk 5 above)
Comment (norgeous) [reply src/scenes/game-scene.ts:161]: would be good for collected coin count

**Review by ScottyXIII — APPROVED** [review]

File: src/scenes/game-scene.ts (hunk 6 above)
Comment (norgeous) [reply src/scenes/game-scene.ts]: done

**Pull request merged**